services/checks/checks.go

```diff
@@ -1092,6 +1093,19 @@ func (s *Service) downloadChecks(ctx context.Context) ([]check.Check, error) {
 
 	resp, err := api.NewRetrievalAPIClient(cc).GetAllChecks(ctx, &api.GetAllChecksRequest{})
 	if err != nil {
+		// If SaaS credentials have become invalid then force a log out so that the next
+		// check download attempt is successfull. Details: https://jira.percona.com/browse/PMM-7965
+		if st, _ := status.FromError(err); st.Message() == "Invalid credentials." {
```

Comment:
Why you rely on the message instead of code? This error should have `codes.Unauthenticated` code.

Reply:
done